docs/getting-started-guides/kubeadm.md

```diff
@@ -277,53 +296,67 @@ See the [list of add-ons](/docs/admin/addons/) to explore other add-ons, includi
-
-However, thanks to the PRs above, `kube-apiserver` works on ARM from the `v1.4.1` release, so make sure you're at least using `v1.4.1` when running on ARM 32-bit
-
-The multiarch flannel daemonset can be installed this way.
+Currently, only the pod network flannel is working on multiple architectures. You can install it this way:
 
     # export ARCH=amd64
     # curl -sSL "https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml?raw=true" | sed "s/amd64/${ARCH}/g" | kubectl create -f -
 
 Replace `ARCH=amd64` with `ARCH=arm` or `ARCH=arm64` depending on the platform you're running on.
 Note that the Raspberry Pi 3 is in ARM 32-bit mode, so for RPi 3 you should set `ARCH` to `arm`, not `arm64`.
 
-## Limitations
+## Cloudprovider integrations (experimental)
 
-Please note: `kubeadm` is a work in progress and these limitations will be addressed in due course.
-Also you can take a look at the troubleshooting section in the [reference document](/docs/admin/kubeadm/#troubleshooting)
+We have got a lot of requests of how to enable a specific cloudprovider. This is possible, but currently requires manual configuration, 
+and therefore not supported. As a first step if you want to do it, edit the `kubeadm` dropin for the `kubelet` service
+(`/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`) on all nodes, including the master. If your cloud provider requires any
+extra packages installed on host, for example for volume mounting/unmounting, install those packages.
 
-1. The cluster created here doesn't have cloud-provider integrations by default, so for example it doesn't work automatically with (for example) [Load Balancers](/docs/user-guide/load-balancer/) (LBs) or [Persistent Volumes](/docs/user-guide/persistent-volumes/walkthrough/) (PVs).
-   To set up kubeadm with CloudProvider integrations (it's experimental, but try), refer to the [kubeadm reference](/docs/admin/kubeadm/) document.
+Specify the `--cloud-provider` flag to kubelet and set it to the cloud of your choice. If your cloudprovider requires a configuration
+file, create the file `/etc/kubernetes/cloud-config` on every node and set the values your cloud requires. Also append
+`--cloud-config=/etc/kubernetes/cloud-config` to the kubelet arguments.
 
-   Workaround: use the [NodePort feature of services](/docs/user-guide/services/#type-nodeport) for exposing applications to the internet.
+Lastly, run `kubeadm init --cloud-provider=xxx` to bootstrap your cluster with cloud provider features.
+
+The reason we're not fully supporting this workflow is that it's very manual process, and it doesn't fit into the rest of the kubeadm
+flow really. But we're working on making it super-easy to spin up a cluster with cloudprovider power-ups. See 
```

Comment:
Suggest: This workflow is not yet fully supported, however we hope to make it extremely easy to spin up clusters with cloud providers in the future. (See this proposal for more information) The Kubelet Dynamic Settings feature may also help to fully automate this process in the future.